### PR TITLE
Minor bugfixes

### PR DESCRIPTION
**Pull request by kbressem**
Fixes two issues: 
1. PyYAML is not compatible with pathlib.Path objects, which were introduced in the config at #19 
2. `if y` in viewer.BasicViewer did not work for arrays. `if y is not None` does work.

### CODE REVIEW

File: trainlib/trainer.py [hunk 1]
```diff
@@ -296,9 +296,17 @@ def _prepare_dirs(self) -> None:
...
-            f.write(yaml.safe_dump(self.config))
+            config = dict(self.config)
+            # convert pathlib.Path to string, because of incompatibility with PyYAML
+            for path in ["run_id", "out_dir", "model_dir", "log_dir"]:
+                config[path] = str(config[path])
+
+            for path in ["data_dir", "train_csv", "valid_csv", "test_csv"]:
+                config["data"][path] = str(config["data"][path])
+
+            f.write(yaml.safe_dump(config))
```
Comment (markus-hinsche) [inline trainlib/trainer.py:309]: Suggestion for readability of the yaml

```suggestion
            f.write(yaml.safe_dump(config, indent=4))
```

File: trainlib/trainer.py [hunk 2]
```diff
@@ -296,9 +296,17 @@ def _prepare_dirs(self) -> None:
         """Set up directories for saving logs, outputs and configs of current training session"""
         # create run_id, copy config file for reproducibility
         run_id = Path(self.config.run_id)
-        run_id.mkdir(exist_ok=True)
+        run_id.mkdir(exist_ok=True, parents=True)
         with open(run_id / "config.yaml", "w+") as f:
-            f.write(yaml.safe_dump(self.config))
+            config = dict(self.config)
+            # convert pathlib.Path to string, because of incompatibility with PyYAML
+            for path in ["run_id", "out_dir", "model_dir", "log_dir"]:
```
Comment (markus-hinsche) [inline trainlib/trainer.py:303]: Hardcoding things that are path: easy to  miss one. 

Instead could we check for `isinstance(config_value, Path)`?

**Review by markus-hinsche — APPROVED** [review]
LGTM

Just to handy suggestions/improvements mentioned

File: trainlib/trainer.py (hunk 2 above)
Comment (kbressem) [reply trainlib/trainer.py:303]: The problem is, that the config is nested. I'll have to write a function that goes recursively through all entries. 
There should be no possibility, that the user adds new paths.

I'll plan to rewrite the config parser to (A) get rid of munch and (B) allow #25 and (C) allow better hyperparameter tuning, directly from the config. I will then implement a better solution to check for paths.

**Pull request merged**